src/python/pants/binaries/binary_tool.py

```diff
@@ -114,6 +118,23 @@ def register_options(cls, register):
     if 'fingerprint' not in version_registration_kwargs:
       version_registration_kwargs['fingerprint'] = True
     register('--version', **version_registration_kwargs)
+    register('--fingerprint', type=str,
+             default=(cls.default_digest.fingerprint if cls.default_digest else None),
+             help='The expected sha256sum of the file to download for this tool. '
+                  'This is only used when a v2 @rule invokes a BinaryToolFetchRequest.')
+    register('--size-bytes', type=int,
+             default=(cls.default_digest.serialized_bytes_length if cls.default_digest else None),
+             help='The expected size of the file to download for this tool. '
+                  'This is only used when a v2 @rule invokes a BinaryToolFetchRequest.')
```

Comment:
Would be great to provide some guidance for how to calculate these values. For example: 

> To calculate the sha256sum for a file, first download it then run `$ sha256 --encrypted ./file.ext`

Reply:
Added a note about how to calculate these. Note that the previous iteration mentioned the specific command, which is literally `sha256sum` -- 0d5bbb0 makes this more explicit!